prediction_market_agent/agents/microchain_agent/nft_treasury_game/app_nft_treasury_game.py

```diff
@@ -240,23 +270,28 @@ def show_about_agent_part(nft_agent: type[DeployableAgentNFTGameAbstract]) -> No
     )
     st.markdown("---")
     with st.popover("Show unprocessed incoming messages"):
-        messages = fetch_unseen_transactions(nft_agent.wallet_address)
+        show_n = 10
+        n_messages = fetch_count_unprocessed_transactions(nft_agent.wallet_address)
+        messages = fetch_unseen_transactions(nft_agent.wallet_address, n=show_n)
```

Comment:
```suggestion
        messages = fetch_unseen_transactions(nft_agent.wallet_address, n=show_n)
        n_messages = len(messages)
```

Reply:
How is retrieving ~400 messages and then showing only 10 of them going to reduce RPC calls in case vs. retrieving 10 messages  and showing 10 messages?

Reply:
Ow, my bad! I was thinking for some reason that we were able to fetch all N (using `fetch_unseen_transactions`) at once, instead of iterating that via `par_map`.
All good here.